### PR TITLE
post and commetn like flow

### PR DESCRIPTION
**Pull request by ravichodvadiya777**

### CODE REVIEW

File: src/controller/commentController.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            }
+        }
+        const like = await likeHelper.insertOne(obj);
+        return global.sendResponse(res, 201, true, "Liked Successfully", like);
+
+    } catch (error) {
+        console.log(error);
+        return global.sendResponse(res, 400, false, "Something not right, please try again.");   
+    }
+}
```
Comment (mohitbaraiya) [inline src/controller/commentController.ts:58]: Like  should have seprate controller

File: src/controller/commentController.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        // await followHelper.delete(obj);
+        
+        // return global.sendResponse(res, 201, true, "Unfollow successfully.");
+    } catch (error) {
+        console.log(error);
+        return global.sendResponse(res, 400, false, "Something not right, please try again.");   
+    }
+}
+
+// ========================================================== End Comemnt Flow ==========================================================
```
Comment (mohitbaraiya) [inline src/controller/commentController.ts:83]: remove un wanted code

File: src/controller/followController.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        const followId = new Types.ObjectId(req.body.follow);
+
+        const obj:FollowType = {
+            user: userId,
+            follow: followId,
+        };
+        
+        const alreadyFollow = await followHelper.findOne(obj);
+        if(alreadyFollow) {return global.sendResponse(res, 409, false, "Already Followed");}
+
```
Comment (mohitbaraiya) [inline src/controller/followController.ts:23]: there should be a middleware to check existing records

File: src/controller/followController.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        const followId = new Types.ObjectId(req.body.follow);
+
+        const obj:FollowType = {
+            user: userId,
+            follow: followId,
+        };
+        const unfollow = await followHelper.findOne(obj);
+        if(!unfollow){
+            return global.sendResponse(res, 409, false, "You can't unfollow");
+        }
```
Comment (mohitbaraiya) [inline src/controller/followController.ts:50]: there should be middleware for record exists or not

File: src/db/commentHelper.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        try {
+            let queryBuilder = Comment.findOne(query);
+            
+            if(select) {
+                queryBuilder = queryBuilder.select(select);
+            }
+            
+            const follow = await queryBuilder.exec();
+            
+            return follow;
```
Comment (mohitbaraiya) [inline src/db/commentHelper.ts:18]: use appropriate name of variable

File: src/db/commentHelper.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        } catch (error) {
+            console.error('Error retrieving users:', error);
```
Comment (mohitbaraiya) [inline src/db/commentHelper.ts:20]: retrieving comments not users

File: src/db/commentHelper.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            throw error;
+        }
+    },
+
+    insertOne: async (data: CommentType) => {
+        try {
+            const result = await Comment.create(data);
+            return result;
+        } catch (error) {
+            console.error('Error adding user:', error);
```
Comment (mohitbaraiya) [inline src/db/commentHelper.ts:30]: comment not user

File: src/db/commentHelper.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            throw error;
+        }
+    },
+
+    delete: async (query: object) => {
+        try {
+            const result = await Comment.deleteOne(query, {new : true});
+            return result;
+        } catch (error) {
+            console.error('Error adding user:', error);
```
Comment (mohitbaraiya) [inline src/db/commentHelper.ts:40]: deleting  comment

File: src/db/followHelper.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+            
+            if(select) {
+                queryBuilder = queryBuilder.select(select);
+            }
+            
+            const follow = await queryBuilder.exec();
+            
+            return follow;
+        } catch (error) {
+            console.error('Error retrieving users:', error);
```
Comment (mohitbaraiya) [inline src/db/followHelper.ts:19]: follow list not users

File: src/middleware/getRecord.ts [hunk 10]
```diff
@@ -8,7 +8,7 @@ export const getRecord = (Model:Model<any>) => {
       return async (req:Request, res: Response, next: NextFunction) => {
         let record;
         try {
-            // console.log(req.params.id);
+          console.log(req.params.id);
```
Comment (mohitbaraiya) [inline src/middleware/getRecord.ts:11]: remove unwanted console

File: src/controller/followController.ts (hunk 3 above)
Comment (ravichodvadiya777) [reply src/controller/followController.ts:23]: this is not possible bcoz i have to 2 filed wise find and id not give params. id get body

**Pull request merged**